hasher-matcher-actioner/hmalib/indexers/LCCIndexer.py

```diff
@@ -0,0 +1,65 @@
+from threatexchange.signal_type.pdq_index import PDQIndex
+from hmalib.common.logging import get_logger
+from datetime import timedelta, datetime
+import time
+from threatexchange.signal_type.timebucketizer import TimeBucketizer, CSViable
+import pickle
+starttime = time.time()
+logger = get_logger(__name__)
+import typing as t
+from dataclasses import dataclass
+
+# @staticmethod
+# def get_file_contents2(
+#         since: datetime.datetime,
+        # until: datetime.datetime,
+        # type: str,
+        # storage_path: str,
+        # bucket_width: datetime.timedelta,
+        # type_class: t.Type[CSViable],
+# ):
+
+Self = t.TypeVar("Self", bound="CSViable")
+
+@dataclass(eq=True)
+class HashRecord(CSViable):
```

Comment:
This PR should only add HashRecord. I want to review LCCIndexer separately. That might be a longer discussion. 

Re-raise this PR (or beat it into shape).

Reply:
Also, move the HashRecord class to a more 'common' place. Both indexers and hashers will use it. So putting it in `hmalib.indexers` is sending the wrong message.

eg. [here](https://github.com/facebook/ThreatExchange/blob/main/hasher-matcher-actioner/hmalib/common/models/pipeline.py)